packages/sdk/client-services/src/packlets/spaces/notarization-plugin.ts

```diff
@@ -180,6 +248,39 @@ export class NotarizationPlugin implements CredentialProcessor {
   setWriter(writer: FeedWriter<Credential>) {
     invariant(!this._writer, 'Writer already set.');
     this._writer = writer;
+    if (this._edgeClient) {
+      this._notarizePendingEdgeCredentials(this._edgeClient, writer);
+    }
+  }
+
+  private _notarizePendingEdgeCredentials(client: EdgeHttpClient, writer: FeedWriter<Credential>) {
+    setTimeout(async () => {
```

Comment:
`scheduleTask(this._ctx, ...)` (make this class a resource)

Every there's a naked setTimeout like this you're risking a race condition: the timeout gets scheduled, and then the class gets closed immediatelly after while the timeout is queued to run.

Reply:
scheduleMicrotask even -- if you don't specifically need it to be a macro task